src/AutoMapper/QueryableExtensions/ProjectionExpression.cs

```diff
@@ -104,7 +105,10 @@ internal class MemberVisitor : ExpressionVisitor
         {
             protected override Expression VisitLambda<T>(Expression<T> node)
             {
-                var memberExpression = node.Body as MemberExpression;
+                MemberExpression memberExpression = 
+                    node.Body.NodeType == ExpressionType.Convert 
+                        ? (node.Body as UnaryExpression)?.Operand  as MemberExpression
+                        :  node.Body as MemberExpression;
```

Comment:
That's replicated code, you need the cast to MemberExpression only once. Also, I'm not sure how we would know any convert expression is about this case. I think we would need more here.